### ISSUE TITLE
Goodtables raises 'field' error for non-matching-header with order_fields=True and infer_fields=False

### ISSUE THREAD

**Issue by flankerqu** [frictionlessdata/goodtables-py#284]
Hello everyone, 

by using goodtables to check one csv file, which:
- covers only part of the columns:
- has extra headers 
- has different column order as defined in data schema
- with order_fields=True and infer_fields=False, 

goodtables will raise an exception 'field' and thus break the validation process.
 
It seems that the error is raised during ordering of the columns.

**Comment by jbothma** [comment]
This is what it looks like for us:

```
Traceback (most recent call last):
  File "bin/run-checks.py", line 53, in <module>
    errors_or_warnings += run_report(filepath)
  File "bin/run-checks.py", line 11, in run_report
    report = run_checks(filepath)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/data_checks/check_datapackage.py", line 21, in run_checks
    order_fields=True
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/validate.py", line 85, in validate
    report = inspector.inspect(source, **options)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/inspector.py", line 83, in inspect
    table_warnings, table_report = task.get()
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 644, in get
    raise self._value
  File "/opt/python/3.6.3/lib/python3.6/multiprocessing/pool.py", line 119, in worker
    result = (True, func(*args, **kwds))
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/inspector.py", line 206, in __inspect_table
    errors += (check_func(header_cells, sample) or [])
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/checks/non_matching_header.py", line 25, in check_headers
    return _check_with_ordering(cells)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/checks/non_matching_header.py", line 51, in _check_with_ordering
    _swap_fields(cells[index], cell)
  File "/home/travis/virtualenv/python3.6.3/lib/python3.6/site-packages/goodtables/checks/non_matching_header.py", line 96, in _swap_fields
    cell1['field'] = cell2['field']
KeyError: 'field'
```

**Comment by roll** [comment]
Fixed in #298
